src/webgpu/util/texture/texelData.ts

```diff
@@ -331,6 +405,25 @@ class TexelDataRepresentationImpl implements TexelDataRepresentation {
     const bytesPerBlock = kUncompressedTextureFormatInfo[this.format].bytesPerBlock;
     assert(!!bytesPerBlock);
 
+    if (this.format === 'rgb9e5ufloat') {
+      assert(this.componentOrder.length === 3);
+      assert(this.componentOrder[0] === TexelComponent.R);
+      assert(this.componentOrder[1] === TexelComponent.G);
+      assert(this.componentOrder[2] === TexelComponent.B);
+      assert(bytesPerBlock === 4);
+      assert(components.R !== undefined);
+      assert(components.G !== undefined);
+      assert(components.B !== undefined);
+
+      const buf = new ArrayBuffer(bytesPerBlock);
+      new DataView(buf).setUint32(
+        0,
+        encodeRGB9E5UFloat(components.R, components.G, components.B),
+        this.isGPULittleEndian
```

Comment:
I think we said this GPU were all going to be little endian?

Reply:
Ah, we can remove this globally in another PR then.